javascript-source/handlers/subscribeHandler.js

```diff
@@ -271,10 +281,84 @@
         }
     });
 
+    /*
+     * @event twitchAnonymousSubscriptionGift
+     */
+    $.bind('twitchAnonymousSubscriptionGift', function (event) {
+        var recipient = event.getRecipient(),
+            months = event.getMonths(),
+            tier = event.getPlan(),
+            message = giftAnonSubMessage;
+
+        if (giftAnonSubWelcomeToggle === true && announce === true) {
+            if (message.match(/\(name\)/g)) {
+                message = $.replace(message, '(name)', 'Anonymous viewer');
```

Comment:
Here you can use `event.getUsername()`, it was added to the core's code, it will return `anonymous`.